modules/asr/asr.h

```diff
@@ -64,11 +64,16 @@ typedef struct asr_attributes_struct
 typedef struct asr_result_struct
 {
     uint16_t id;             ///< Keyword or command ID
+
+    // The following fields are optional and may not be supported by all ASR ports
     uint16_t score;          ///< The confidence score of the detection
     uint16_t gscore;         ///< The garbage score
     int32_t  start_index;    ///< The audio sample index that corresponds to the start of the utterance
     int32_t  end_index;      ///< The audio sample index that corresponds to the end of the utterance
     int32_t  duration;       ///< THe length of the utterance in samples
+    uint32_t sg_diff;        ///< The voice similarity of the detection
+    uint32_t energy;         ///< The energy of the detection
```

Comment:
You have included an element here for the detection energy but it isn't set in DSpotter_asr.c.  Is that your intention?

Reply:
That field was actually not stored in the original code, I fixed that in https://github.com/xmos/sln_voice/commit/795ac29743ceb59813f06646c5eefeda3d4ed274.